Sources/Datadog/Tracing/TracingAutoInstrumentation.swift

```diff
@@ -0,0 +1,103 @@
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal class TracingAutoInstrumentation {
+    static var instance: TracingAutoInstrumentation?
+
+    let tracedHosts: Set<URL>
+    let swizzler: URLSessionSwizzler
+
+    init?(tracedHosts: Set<URL>?) {
+        guard let someHosts = tracedHosts else {
+            return nil
+        }
+        do {
+            self.tracedHosts = someHosts
+            self.swizzler = try URLSessionSwizzler()
+        } catch {
+            userLogger.warn("🔥 Network requests won't be traced automatically for \(String(describing: tracedHosts)): \(error)")
+            developerLogger?.warn("🔥 Network requests won't be traced automatically for \(String(describing: tracedHosts)): \(error)")
+            return nil
+        }
+    }
+
+    func apply() {
+        let interceptor = TracingRequestInterceptor.build(with: tracedHosts)
+        swizzler.swizzle(using: interceptor)
+    }
+}
+
+internal enum TracingRequestInterceptor {
+    static func build(with tracedHosts: Set<URL>) -> RequestInterceptor {
+        let interceptor: RequestInterceptor = { urlRequest in
+            guard let tracer = Global.sharedTracer as? DDTracer,
+                let someURL = urlRequest.url,
+                tracedHosts.allows(someURL),
+                HTTPHeadersWriter.canInject(to: urlRequest) else {
+                    return nil
+            }
+            let spanContext = tracer.createSpanContext()
+            let headersWriter = HTTPHeadersWriter()
+            headersWriter.inject(spanContext: spanContext)
+            let tracingHeaders = headersWriter.tracePropagationHTTPHeaders
+            var modifiedRequest = urlRequest
+            tracingHeaders.forEach { modifiedRequest.setValue($1, forHTTPHeaderField: $0) }
+
+            let operationName = someURL.absoluteString
+            let observer: TaskObserver = tracingTaskObserver(tracer: tracer, spanContext: spanContext, operationName: operationName)
+            return InterceptionResult(modifiedRequest: modifiedRequest, taskObserver: observer)
+        }
+        return interceptor
+    }
+
+    static func tracingTaskObserver(
+        tracer: DDTracer,
+        spanContext: DDSpanContext,
+        operationName: String
+    ) -> TaskObserver {
+        var startedSpan: OTSpan? = nil
+        let observer: TaskObserver = { observedEvent in
+            switch observedEvent {
+            case .starting(let request):
+                if let ongoingSpan = startedSpan {
+                    userLogger.warn("\(String(describing: request)) is starting a new trace but it's already started a trace before: \(ongoingSpan)")
+                    developerLogger?.warn("\(String(describing: request)) is starting a new trace but it's already started a trace before: \(ongoingSpan)")
+                }
+                let span = tracer.startSpan(
+                    spanContext: spanContext,
+                    operationName: operationName
```

Comment:
To unify it with the way network `spans` [are named in Android SDK](https://github.com/DataDog/dd-sdk-android/blob/d512922887ac0d214bfc0b5808ac05c226f90235/dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/net/TracingRequestInterceptor.kt#L57), the `operationName` should be constant: `"urlsession.request"`.

Also, we should set the `resourceName` to `request.url` as [we do in Android SDK](https://github.com/DataDog/dd-sdk-android/blob/d512922887ac0d214bfc0b5808ac05c226f90235/dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/net/TracingRequestInterceptor.kt#L63).

Reply:
```swift
let span = tracer.startSpan(
  spanContext: spanContext,
  operationName: "urlsession.request"
)
span.setTag(key: DDTags.resource, value: url)
```
✅